velero/schedule/keycloak-br-script-cm.yaml

```diff
@@ -0,0 +1,120 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: keycloak-br-configmap
+  labels:
+    foundationservices.cloudpak.ibm.com: keycloak-data
+data:
+  br_keycloak.sh: |
+    #!/usr/bin/env bash
+
+    # Licensed Materials - Property of IBM
+    # Copyright IBM Corporation 2023. All Rights Reserved
+    # US Government Users Restricted Rights -
+    # Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+    #
+    # This is an internal component, bundled with an official IBM product.
+    # Please refer to that particular license for additional information.
+
+    # ---------- Command arguments ----------
+
+    set -o errtrace
+    set -o errexit
+
+    MODE=$1
+    KEYCLOAK_NAMESPACE=$2
+    OPERATOR_NAMESPACE=$3
+    if [[ -z $OPERATOR_NAMESPACE ]]; then
+        OPERATOR_NAMESPACE=$KEYCLOAK_NAMESPACE
+    fi
+
+    BACKUP_DIR=/keycloak/keycloak-backup
+
+    function main {
+        if [[ $MODE == "backup" ]]; then
+            info "Mode set to backup, beginning backup process."
+            backup
+            success "Backup completed successfully."
+        else
+            info "Mode is set to restore, beginning restore process."
+            restore
+            success "Restore completed successfully."
+        fi
+    }
+
+    function backup {
```

Comment:
No need to quiesce the database or enter EDB operator maintenance mode?

Reply:
The script is based on the steps used by zen 5 (private repo, I'll send via slack) and after discussing with Rahul, it's my understanding that we just need to make sure there are no active users. This is why we have scaled down the keycloak operator. This is something I will need to look more into though to be sure.